### PR TITLE
YdbSpliterator bugfix

### PR DESCRIPTION
**Pull request by lavrukov**

### CODE REVIEW

**Comment by lavrukov** [comment]
> Please fix it in repository-ydb-**v2** as well.

Of course.

Please don't merge it, I will add some tests

File: repository-ydb-v1/src/main/java/tech/ydb/yoj/repository/ydb/YdbRepositoryTransaction.java [hunk 1]
```diff
@@ -101,10 +101,10 @@ public YdbRepositoryTransaction(REPO repo, @NonNull TxOptions options) {
         this.cache = options.isFirstLevelCache() ? new RepositoryCacheImpl() : RepositoryCache.empty();
     }
 
-    private <V> Stream<V> makeStream(YdbSpliterator<V> spliterator) {
-        Stream<V> stream = spliterator.makeStream();
-        openedStreams.add(stream);
-        return stream;
+    private <T> YdbSpliterator<T> createSpliterator(String request, boolean isOrdered) {
+        YdbSpliterator<T> spliterator = new YdbSpliterator<>(request, isOrdered);
+        spliterators.add(spliterator);
```
Comment (lavrukov) [inline repository-ydb-v1/src/main/java/tech/ydb/yoj/repository/ydb/YdbRepositoryTransaction.java:106]: It's better for understanding. -1 hop

File: repository-ydb-v1/src/main/java/tech/ydb/yoj/repository/ydb/YdbSpliterator.java [hunk 2]
```diff
@@ -116,24 +116,33 @@ private QueueValue<V> poll() {
     @Override
     // (stream thread) called from stream engine for get new value.
     public boolean tryAdvance(Consumer<? super V> action) {
```
Comment (lavrukov) [inline repository-ydb-v1/src/main/java/tech/ydb/yoj/repository/ydb/YdbSpliterator.java:118]: question: What should we do if spliterator closed ?

**Comment by lavrukov** [comment]
> > Please fix it in repository-ydb-**v2** as well.
> 
> Of course.
> 
> Please don't merge it, I will add some tests

I'm done

File: repository-ydb-v1/src/main/java/tech/ydb/yoj/repository/ydb/YdbSpliterator.java (hunk 2 above)
Comment (nvamelichev) [reply repository-ydb-v1/src/main/java/tech/ydb/yoj/repository/ydb/YdbSpliterator.java:118]: I think we should just return `false` from `tryAdvance`. Java's `Spliterator` interface declares no exceptions except for NPE than can be thrown from `Spliterator.tryAdvance()`

**Review by nvamelichev — APPROVED** [review]
LGTM 👍

**Pull request merged**